### PR TITLE
Feat: deposit auction

### PR DESCRIPTION
**Pull request by haythemsellami**
# Task: deposit auction

## Description

Fixes ENG-1365

## Type of change

- [X] New feature
- [ ] Bug fix
- [ ] Testing code
- [ ] Document update or config files

### CODE REVIEW

File: packages/zen-bull-netting/src/ZenBullNetting.sol [hunk 1]
```diff
@@ -484,10 +545,207 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
         withdrawsIndex = i;
     }
 
+    /**
+     * @notice auction for queued deposits
+     * @dev takes in orders from MM's to buy oSQTH
```
Comment (alpinechicken) [inline packages/zen-bull-netting/src/ZenBullNetting.sol:550]: *MMs

File: packages/zen-bull-netting/src/ZenBullNetting.sol [hunk 2]
```diff
@@ -484,10 +545,207 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
...
+            } else {
+                IWETH(weth).transferFrom(
+                    _params.orders[i].trader,
+                    address(this),
+                    (_params.orders[i].quantity * _params.clearingPrice) / 1e18
+                );
+                oSqthToMint -= _params.orders[i].quantity;
+            }
+        }
+        require(oSqthToMint == 0, "ZBN23");
```
Comment (alpinechicken) [inline packages/zen-bull-netting/src/ZenBullNetting.sol:593]: Why wouldn't we get this?

File: packages/zen-bull-netting/src/ZenBullNetting.sol [hunk 3]
```diff
@@ -484,10 +545,207 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
...
+                break;
+            }
+        }
+
+        // send ZenBull to depositor
+        memVar.remainingDeposits = _params.depositsToProcess;
+        uint256 k = depositsIndex;
+        memVar.currentZenBullBalance =
+            IERC20(zenBull).balanceOf(address(this)) - initialZenBullBalance;
+        memVar.remainingEth =
```
Comment (alpinechicken) [inline packages/zen-bull-netting/src/ZenBullNetting.sol:673]: Already have this on 628

File: packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+        IWETH(WETH).deposit{value: 1000e18}();
+        vm.prank(user2);
+        IWETH(WETH).deposit{value: 1000e18}();
+        vm.prank(mm1);
+        IWETH(WETH).deposit{value: 1000e18}();
+        vm.prank(mm2);
+        IWETH(WETH).deposit{value: 1000e18}();
+    }
+
+    function _calAuctionCrabAmount(uint256 _depositToProcess) internal view returns (uint256) {
```
Comment (alpinechicken) [inline packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol:59]: *_calcAutionCrabAmount?

File: packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+        zenBullNetting.depositAuction(params);
+        vm.stopPrank();
+
+        uint256 wPowerPerpTotalSupplyAfter = IERC20(WPOWERPERP).totalSupply();
+
+        assertEq(
+            IERC20(WPOWERPERP).balanceOf(mm1) - mm1WpowerPerpBalanceBefore,
+            wPowerPerpTotalSupplyAfter - wPowerPerpTotalSupplyBefore
+        );
+        assertGt(user1.balance, user1EthBalanceBefore);
```
Comment (alpinechicken) [inline packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol:253]: Can we test assertApprox for user balance?

File: packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol [hunk 6]
```diff
@@ -0,0 +1,582 @@
...
+        vm.startPrank(owner);
+        zenBullNetting.depositAuction(params);
+        vm.stopPrank();
+
+        uint256 wPowerPerpTotalSupplyAfter = IERC20(WPOWERPERP).totalSupply();
+
+        assertEq(
+            IERC20(WPOWERPERP).balanceOf(mm1) - mm1WpowerPerpBalanceBefore,
+            wPowerPerpTotalSupplyAfter - wPowerPerpTotalSupplyBefore
+        );
```
Comment (alpinechicken) [inline packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol:252]: test crab balances?

File: packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol [hunk 7]
```diff
@@ -0,0 +1,582 @@
...
+        uint256 expectBullAmount = _depositToProcess * ethUsdPrice * 99e16 / 1e18 / bullEquityValue
+            * IZenBullStrategy(ZEN_BULL).totalSupply() / 1e18;
+
+        return expectBullAmount * 1e18 / IZenBullStrategy(ZEN_BULL).totalSupply()
+            * IZenBullStrategy(ZEN_BULL).getCrabBalance() / 1e18;
+    }
+
+    function testFullDepositAuction() public {
+        uint256 amount = 10e18;
+        _queueEth(user1, amount);
```
Comment (alpinechicken) [inline packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol:84]: Maybe also test with more than one user queuing.

File: packages/zen-bull-netting/src/ZenBullNetting.sol (hunk 3 above)
Comment (haythemsellami) [reply packages/zen-bull-netting/src/ZenBullNetting.sol:673]: Yeah but could change after the flashdeposit

File: packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol (hunk 4 above)
Comment (haythemsellami) [reply packages/zen-bull-netting/test/integration-test/ZenBullNetting/DepositAuction.t.sol:59]: ?

**Pull request merged**